### PR TITLE
So many changes!

### PR DESCRIPTION
**Pull request by sanbox-irl**
ShaderLanguageType, PseudoBoolean, SoundKind, SoundType,
PathKind, EventType, EventNumber.

Added interfaces for:
Rooms, Layer, Instance, Colour, PhysicsSettings, roomSettings,
View, viewSettings, Path, PathPoint, GMFolder,
Tileset (incomplete), Font, Glyph, Timeline, Moment,
Note, Extension.

Updated:
Object, ObjectEvent, Sound, Shader

Minor changes elsewhere / I forgot what else I did, do not
kill me with the famous Iron Fist of topher.

Looking at it, there many be some formatting issues. Let me know exactly how you want that if you see any issues and I'll resend it.

Thanks for the basics!

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -32,189 +171,526 @@ declare namespace Resource {
...
+      inheritPhysicsSettings: boolean;
+      modelName: string;
+      PhysicsWorld: boolean;
+      PhysicsWorldGravityX: number;
+      PhysicsWorldGravityY: number;
+      PhysicsWorldPixToMeters: number;
+      mvc: string;
+  }
+
+  export interface roomSettings {
```
Comment (christopherwk210) [inline index.d.ts:355]: Unless there's a specific reasoning behind it, interfaces should all be PascalCase

File: index.d.ts [hunk 2]
```diff
@@ -32,189 +171,526 @@ declare namespace Resource {
...
+      vspeed: number;
+      wport: number;
+      wview: number;
+      xport: number;
+      xview: number;
+      yport: number;
+      yview: number;
+  }
+
+  export interface viewSettings {
```
Comment (christopherwk210) [inline index.d.ts:386]: Another interface that should be PascalCase, unless intentional?

File: index.d.ts [hunk 3]
```diff
@@ -287,17 +760,17 @@ interface YYP {
   /** Unknown property, seems to always be an empty array */
   configs: Array<any>;
 
-  /** Unknown property, seems to always be false */
+  /** Allows for experimental JS editing. Unfinished or legacy feature. It's a secret. */
```
Comment (christopherwk210) [inline index.d.ts:763]: Thanks for updating this, love it!

**Review by christopherwk210 — COMMENTED** [review]
I found two interfaces that weren't PascalCase, I commented on each of them. Is there a specific reasoning behind that?

**Comment by sanbox-irl** [comment]
Yup, all of those are camelCase in the YYs for some reason. Absolutely no clue what's up with that.

File: index.d.ts (hunk 3 above)
Comment (sanbox-irl) [reply index.d.ts:763]: No prob!

**Pull request merged**